lib/opencensus/trace/span_builder.rb

```diff
@@ -123,17 +146,149 @@ def finish!
+      #     unsigned `(0..2^64-1)`
+      # @param [Integer] uncompressed_size The number of uncompressed bytes
+      #     sent or received.
+      # @param [Integer] compressed_size The number of compressed bytes sent
+      #     or received. (optional, defaults to the same as uncompressed_size)
+      # @param [Time] time Timestamp of the event (optional, defaults to the
+      #     current time)
+      #
+      def put_message_event type, id, uncompressed_size,
+                            compressed_size: 0, time: nil
+        time ||= Time.now.utc
+        message_event = MessageEvent.new time, type, id, uncompressed_size,
+                                         compressed_size
+        @message_events << message_event
+        self
+      end
+
+      ##
+      # Add a pointer from the current span to another span, which may be in
+      # the same trace or in a different trace. For example, this can be used
+      # in batching operations, where a single batch handler processes multiple
+      # requests from different traces or when the handler receives a request
+      # from a different project.
+      #
+      # @param [String] trace_id The unique identifier for a trace. A 16-byte
+      #     array expressed as 32 hex digits.
+      # @param [String] span_id The unique identifier for a span within a trace.
+      #     An 8-byte array expressed as 16 hex digits.
+      # @param [Symbol] type The relationship of the current span relative to
+      #     the linked span. Valid values are `SpanBuilder::CHILD_LINKED_SPAN`,
+      #     `SpanBuilder::PARENT_LINKED_SPAN`, and
+      #     `SpanBuilder::TYPE_UNSPECIFIED`.
+      # @param [String] attributes Key-value pairs providing additional
+      #     properties of the link. Keys must be strings, and values are
+      #     restricted to the same types as attributes (see #put_attribute).
+      #
+      def put_link trace_id, span_id, type, attributes = {}
+        link = Link.new trace_id, span_id, type, attributes
+        @links << link
+        self
+      end
+
+      ##
+      # Set the optional final status for the span.
+      #
+      # @param [Integer] code Status code as a 32-bit signed integer
+      # @param [String] message A developer-facing error message, which should
+      #     be in English.
+      def set_status code, message = ""
+        @status_code = code
+        @status_message = message
+        self
+      end
+
+      ##
+      # Set the stack trace for this span.
+      # You may call this in one of three ways:
+      #
+      # *   Pass in no argument to use the caller's stack trace.
+      # *   Pass in an integer to use the caller's stack trace, but skip
+      #     additional stack frames.
+      # *   Pass in an explicit array of Thread::Backtrace::Location as
+      #     returned from Kernel#caller_locations
+      #
+      # @param [Array{Thread::Backtrace::Location}, Integer] locations
```

Comment:
Looks like for YARD, arrays are `Array<subtype>`.

Reply:
Okay, fixing.